feat: add project logo

Megrax opened the pull request:
Display logo of repo's organization/owner before the chart title.
<img width="600" src="https://user-images.githubusercontent.com/56376387/173192658-bdec46c4-e7f6-4aaf-8360-2cd5c22be6ea.png"></img>

If there're multiple repos of multiple owners, logos will be displayed in the legend, right before each repo's name.
<img width="600" src="https://user-images.githubusercontent.com/56376387/173192737-5d9bbc3f-8a77-4b13-bfd5-b9c80068c940.png"></img>

But if all repos have the same owner, logo will still be displayed before the title.
<img width="600" src="https://user-images.githubusercontent.com/56376387/173192681-b0765ff9-4e59-435c-8b69-07a1350de750.png"></img>

Closes #168

boojack commented on packages/xy-chart/utils/drawLabels.ts:
```diff
@@ -14,30 +15,46 @@ export const drawTitle = (
+    .attr("id", "clip-circle-title")
+    .append("circle")
+    .attr("r", 11)
+    .attr("cx", "39.5%")
```
Relative positions are not suitable for widescreen. Maybe we can calculate the width of `d3.selection` first and use absolute position instead. 

<img width="417" alt="image" src="https://user-images.githubusercontent.com/24653555/173199397-b292c5eb-483c-42d4-ac7d-b579fe8df267.png">

But in server runtime(node.js), we can't call any document element methods. So it needs to take a deep look for a better solution.

Megrax commented:
Updated. Added support for various server side sizes also.

Mobile:
<img src="https://user-images.githubusercontent.com/56376387/173560594-575c3afe-ab6d-4fb8-bd53-aab2a5c45200.png" width="500"></img>

Laptop:
<img src="https://user-images.githubusercontent.com/56376387/173560668-6a4327a5-5291-4e40-b4c8-0a2b6e0d5e18.png" width="500"></img>

Desktop:
<img src="https://user-images.githubusercontent.com/56376387/173560726-11138282-8783-47c3-9d25-140d761158ff.png" width="500"></img>

boojack commented on common/chart.ts:
```diff
@@ -67,6 +67,120 @@ export const getReposStarData = async (
+      let message = "";
+      let status = 500;
+
+      if (error?.response?.status === 404) {
```
Let's remove the switch-like handler for response error when fetching repo logoUrl.

Megrax replied:
Actually this `getReposLogoUrl` method is not used anywhere. I've made getting both star records and logo url into one `getRepoData` method. So I can just remove this whole `getReposLogoUrl` method.

boojack replied:
Then let's remove it.

boojack commented on packages/xy-chart/index.ts:
```diff
@@ -69,6 +71,8 @@ export interface XYChartOptions {
   backgroundColor: string;
   strokeColor: string;
+
+  serverWidth?: number;
```
How about call it `chartWidth`?

Megrax replied:
I named it `serverWidth` because it's only used for server side configured width (`mobile`, `laptop`, `desktop`). How 
 about `serverChartWidth`?

Megrax replied:
After more thinking, `chartWidth` is OK because it's more general, and I'll change it.

boojack replied:
Frond-end could use this field too I think. In the embeded star-history, we could also set a special chart width. 

<img width="1097" alt="image" src="https://user-images.githubusercontent.com/24653555/173728464-93011b62-cb92-43c7-9188-30a2789905d7.png">

And it's better not to have some special target platform fields in `packages/xy-chart`. Make it purer.

boojack reviewed (changes requested):
Some nits

Megrax commented:
PTAL @boojack

boojack reviewed (approved):
LGTM. Nice work~

![star-history-2022615 (2)](https://user-images.githubusercontent.com/24653555/173730984-32239b66-449b-4884-bcdc-6657db504bf4.png)

dragonly commented:
> But if all repos have the same owner, logo will still be displayed before the title.

Love this UX :smile:
